src/components/taskDetails/Details.tsx

```diff
@@ -5,16 +5,68 @@ import { TaskDetailsProps } from '@/interfaces/taskDetails.type';
 import extractRepoName from '@/utils/extractRepoName';
 import Link from 'next/link';
 import { FaReceipt } from 'react-icons/fa6';
-import DevFeature from '../DevFeature';
+import moment from 'moment';
+import Tooltip from '@/components/common/Tooltip/Tooltip';
+import { useRouter } from 'next/router';
 
 const Details: FC<TaskDetailsProps> = ({ detailType, value, url }) => {
     const color = value ? setColor?.[value] : undefined;
     const isGitHubLink = detailType === 'Link';
     const gitHubIssueLink = isGitHubLink ? value : undefined;
+    const router = useRouter();
+    const isDevMode = router.query.dev === 'true' ? true : false;
+
+    const getRelativeTime = (
+        timestamp: string | number | undefined
+    ): string => {
+        return timestamp ? moment(timestamp).fromNow() : 'N/A';
+    };
+
+    const getTooltipText = (timestamp: string | number | undefined): string => {
+        return timestamp ? moment(timestamp).format('LLLL') : 'N/A';
+    };
+
+    const isPastDate = (timestamp: string | number | undefined): boolean => {
+        return timestamp ? moment(timestamp).isBefore(moment()) : false;
+    };
+    /**
+     * Get human-readable time format from the provided timestamp.
+     * @param {string | number | undefined} timestamp - The timestamp.
+     * @returns {string} - The human-readable time format, for example: instead of displaying a timestamp like "2024-02-20T14:30:00", a human-readable format would present it as "February 20, 2024, 2:30 PM
+     */
+    const getHumanReadableTime = (
+        timestamp: string | number | undefined
+    ): string => {
+        if (!timestamp) return 'N/A';
+
+        const now = moment();
+        const futureTimestamp = moment(timestamp);
+
+        if (futureTimestamp.isAfter(now)) {
+            return `in ${moment
+                .duration(futureTimestamp.diff(now))
+                .humanize()}`;
+        }
+        return `${moment.duration(now.diff(futureTimestamp)).humanize()} ago`;
+    };
+
+    const isTimeDetail =
+        detailType === 'Started On' || detailType === 'Ends On';
+
+    const renderedValue = value ?? 'N/A';
+    const formattedDetailType = isDevMode
+        ? detailType === 'Started On'
+            ? 'Started'
+            : detailType === 'Ends On'
+            ? 'Ended'
+            : detailType
+        : detailType;
+
+    const shouldRenderTooltip = isTimeDetail && isDevMode;
```

Comment:
Your condition will also work but here's my NIT suggestion

```suggestion
    const shouldRenderTooltip = isDevMode && isTimeDetail ;
```

Reason: since if devMode is not true it doesn't checks the isTimeDetail and moves on.

Reply:
Done. Please re-check